lib/presentation/view/details.dart

```diff
@@ -0,0 +1,447 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_html/flutter_html.dart';
+import 'package:get/get.dart';
+
+import '../../core/util/asset_constants.dart';
+import '../../core/util/dimensions.dart';
+import '../../core/util/palette.dart';
+import '../../core/util/text_constants.dart';
+import '../controller/ingredients_controller.dart';
+import '../controller/nutritional_value_controller.dart';
+import '../controller/recipe_controller.dart';
+import '../controller/similar_recipes_controller.dart';
+import '../widget/company_logo.dart';
+import '../widget/footer.dart';
+import '../widget/ingredient_card.dart';
+import '../widget/recipe_card_vertical.dart';
+import '../widget/recipe_flags.dart';
+import '../widget/section_title.dart';
+
+class Details extends StatelessWidget {
+  static const double similarRecipeWidth = 200;
+  static const double gridPadding = 20;
+  static const double gridMaxCrossAxisExtent = 250;
+  static const double ingredientCrossAxisExtent = 200;
+  static const double gridSpacing = 10;
+  static const int similarRecipesTitleMaxLines = 1;
+  static const String idParameterField = 'id';
+  static const double gridChildAspectRatio = .8;
+  static const String noSimilarRecipesMessage = 'No similar recipes found';
+
+  final RecipeController recipeController = Get.find<RecipeController>();
+  final NutritionalValueController nutritionalValueController =
+      Get.find<NutritionalValueController>();
+  final IngredientsController ingredientsController =
+      Get.find<IngredientsController>();
+  final SimilarRecipesController similarRecipesController =
+      Get.find<SimilarRecipesController>();
+
+  Details({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Palette.mainBackground,
+      appBar: AppBar(
+        backgroundColor: Palette.primary,
+        automaticallyImplyLeading: false,
+        title: const CompanyLogo(),
+        actions: [
+          Padding(
+            padding: EdgeInsets.symmetric(
+              horizontal: Get.width * Dimensions.fivePercent,
+            ),
+            child: Image.asset(
+              AssetConstants.searchIcon,
+              width: Dimensions.searchIconWidth,
+              color: Colors.white,
+              filterQuality: FilterQuality.medium,
+            ),
+          ),
+        ],
+      ),
+      body: ListView(
```

Comment:
To be more readable could you separated the creation of the details in diferent kind of builders like _buildHeader, buildNutritionalInfo, _buildRecipeIngredients

Reply:
Created separated builders